client/src/pages/dashboards/BoardDashboard.js

```diff
@@ -0,0 +1,336 @@
+          setEditedNode(
+            model.getSelectedEntities().length === 1
+              ? model.getSelectedEntities()[0]
+              : null
+          )
+      }
+      model.registerListener({
+        nodesUpdated: function(event) {
+          if (event.isCreated) {
+            event.node.registerListener(selectionListener)
+          } else {
+            if (
+              editedNode?.options.anetObject.uuid ===
+              event.node.anetObject?.uuid
+            ) {
+              setEditedNode(null)
+            }
+          }
+        }
+      })
+      model.getNodes().forEach(node => node.registerListener(selectionListener))
+    }
+  }, [editedNode, model])
+
+  useEffect(() => {
+    model && model.setLocked(!edit)
+  }, [model, edit])
+
+  useEffect(() => {
+    async function fetchData() {
+      await fetch(dashboardSettings.data)
+        .then(response => response.json())
+        .then(data => {
+          const model = new DiagramModel()
+          model.deserializeModel(data, engineRef.current)
+          setModel(model)
+        })
+    }
+    fetchData()
+  }, [dashboardSettings.data])
+
+  useEffect(() => {
+    if (dropEvent) {
+      const { data, point } = dropEvent
+      const node = new DiagramNodeModel(data)
+
+      node.setPosition(point)
+      model.getSelectedEntities().forEach(entity => entity.setSelected(false))
+      node.setSelected(true)
+      setEditedNode(node)
+      engineRef.current.getModel().addNode(node)
+      setDropEvent(null)
+    }
+  }, [model, dropEvent])
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "row",
+        height: "100%"
+      }}
+    >
+      <div
+        style={{
+          position: "relative",
+          flexGrow: 1
+        }}
+        onDrop={event => {
+          event.persist()
+          const data = utils.parseJsonSafe(
+            event.dataTransfer.getData("storm-diagram-node")
+          )
+          const point = engineRef.current.getRelativeMousePoint(event)
+          setDropEvent({ data, point })
+        }}
+        onDragOver={event => {
+          event.preventDefault()
+        }}
+      >
+        {engineRef.current.getModel() && (
+          <CanvasWidget
+            engine={engineRef.current}
+            className="diagram-container"
+          />
+        )}
+      </div>
+      <div>
+        <Panel>
+          <Button
+            bsStyle="primary"
+            type="button"
+            onClick={() => setEdit(!edit)}
+          >
+            {edit ? <Icon icon={IconNames.DOUBLE_CHEVRON_RIGHT} /> : "Edit"}
+          </Button>
+
+          {edit && (
+            <>
+              <Button onClick={() => engineRef.current.zoomToFit()}>
+                <Icon icon={IconNames.ZOOM_TO_FIT} />
+              </Button>
+              <Button
+                onClick={() => {
+                  dagreEngineRef.current.redistribute(model)
+
+                  // this.reroute();
```

Comment:
Why is this commented out (perhaps add a reason/TODO)?

Reply:
@VassilIordanov can you answer this one?